text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+
+In this example, the instrumentation should emit:
+
+- The current per-core CPU usage (_agg[core]_)
+- The total CPU usage of each job, across all cores and instances (_agg[job]_)
+
+| time | core | job | value | agg([core])                               | agg([job])                               | agg([core, job])                                                                                                         |
+|------|------|-----|-------|-------------------------------------------|------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
+| 1    | 1    | A   | 1     | `({core: 1}, 1)`                          | `({job: A}, 1)`                          | `({core: 1, job: A}, 1)`                                                                                                 |
+| 2    | 2    | A   | 10    | `({core: 1}: 1)` <br> `({core: 2}: 10)`   | `({job: A}, 11)`                         | `({core: 1, job: A}, 1)` <br> `({core: 2, job: A}, 10)`                                                                  |
+| 3    | 1    | B   | 100   | `({core: 1}: 100)` <br> `({core: 2}: 10)` | `({job: A}, 11)` <br> `({job: B}, 100)`  | `({core: 1, job: A}, 1)` <br> `({core: 2, job: A}, 10)` <br> `({core: 1, job: B}, 100)`                                  |
+| 4    | 2    | B   | 1000  | `({core: 1}: 1)` <br> `({core: 2}: 1000)` | `({job: A}, 11)` <br> `({job: B}, 1100)` | `({core: 1, job: A}, 1)` <br> `({core: 2, job: A}, 10)` <br> `({core: 1, job: B}, 100)` <br> `({core: 2, job: B}, 1000)` |
+
+Here the aggregated value of _agg([job])_ depends on the value of the "core" label, even though it's not preserved in the aggregation: it is the sum over the last-reported value for each core.
+
+This makes configuring views for Observers significantly more complicated than for Measures.
+
+This OTEP does not attempt to solve this problem.
+We propose to make LastValue the only valid aggregation for Observer instruments, and leave it up to the user to specify label keys that produce coherent metrics for each registered Observer.
+
+## Prior art and alternatives
+
+One alternative is not to include a view API, and require users to configure metric instruments and aggregations individually.
+This approach is partially described in the current spec.
+See the motivation section for an argument against this approach.
+
+This API borrows heavily from [OpenCensus views](https://opencensus.io/stats/view/).
+
+See also [DataDog metric types](https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#metric-types) and [Prometheus metric types](https://prometheus.io/docs/concepts/metric_types).
+
+## Open questions
+
+Should we ignore calls to record measurements from instruments that don't appear in any view?
+This is potentially a nice optimization, but it would mean that no metrics are exported by default from instrumented code.
```

Comment:
The alternative is to keep using default aggregation types if no view is defined, and then the views would take priority if defined.